tests/console/ovs_server.pm

```diff
@@ -0,0 +1,156 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test IPsec tunnel in Open vSwitch with 3 different authentication methods
+#
+#   This test does the following
+#    - Installs openvswitch-ipsec and openvswitch-pki
+#    - Starts the systemd service unit
+#    - Executes IPsec tunneling between two hosts with the following authenitcation
+#    methods:
+#    	* Pre-shared key
+#    	* Self-signed certificate
+#    	* CA-signed certificate
+#    - This host verifies that IPsec GRE tunnel is running between the two hosts
+#
+# Maintainer: Anna Minou <anminou@suse.de>
+#
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use utils;
+use console::ovs_utils;
+
+my $server_ip   = "10.0.2.101";
+my $client_ip   = "10.0.2.102";
+my $id          = "1";
+my $dir         = "/etc/keys/";
+my $dir_certs   = "/etc/ipsec.d/certs/";
+my $dir_private = "/etc/ipsec.d/private/";
+my $dir_cacerts = "/etc/ipsec.d/cacerts/";
+
+sub run {
+
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    # Install the needed packages
+    zypper_call('in openvswitch-ipsec tcpdump openvswitch-pki', timeout => 300);
+    systemctl 'start openvswitch';
+    systemctl 'start openvswitch-ipsec';
+
+    # Setup the ovs bridge
+    add_bridge("$id");
+
+    # Setup IPsec tunnel using pre-shared keys
+    assert_script_run("ovs-vsctl add-port br-ipsec tun -- set interface tun type=gre options:remote_ip=$client_ip options:psk=swordfish");
+
+    # Wait for the client host to setup the IPsec tunnel
+    barrier_wait 'ipsec_done';
+
+    # Check that ESP packets are being sent from this host(server) to the other(client)
+    assert_script_run("curl -O " . data_url('console/ping.sh'));
```

Comment:
Why not just run ping in one `assert_script_run` ?

Reply:
I didn't manage to run it in the background using '&' in the end of the command with `assert_script_run` . It fails :/

Reply:
Like this ? https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/sshd.pm#L105

Reply:
Thanks for the hint Jozef, it is updated now :)